nuclia/e2e/conftest.py

```diff
@@ -0,0 +1,300 @@
+    "phone": "+34 111 222 333",
+    "receive_updates": False
+}
+
+
+@pytest.fixture(scope="session")
+def test_config():
+    return {
+        "test_account_slug": TEST_ACCOUNT_SLUG
+    }
+
+
+class ManagerAPI:
+    def __init__(self, global_api):
+        self.global_api = global_api
+        self.session = requests.Session()
+
+    def delete_account(self, account_slug) -> bool:
+        url = f"{self.global_api.base_url}/api/manage/@account/{account_slug}"
+        response = self.session.delete(url, headers=self.global_api.root_auth_headers)
+        if response.status_code not in (204, 404):
+            response.raise_for_status()
+        return response.status_code == 204
+
+class GlobalAPI:
+    def __init__(self, base_url, recaptcha, root_pat_token):
+        self.base_url = base_url
+        self.session = requests.Session()
+        self.recaptcha = recaptcha
+        self.access_token = None
+        self.root_pat_token = root_pat_token
+        self.manager = ManagerAPI(self)
+
+    @property
+    def auth_headers(self):
+        headers = {
+            'Authorization': f'Bearer {self.access_token}',
+        }
+        return headers
+
+    @property
+    def root_auth_headers(self):
+        headers = {
+            'Authorization': f'Bearer {self.root_pat_token}',
+        }
+        return headers
+
+    def signup(self, name, email, password):
+        url = f"{self.base_url}/api/auth/signup"
+        payload = {
+            "name": name,
+            "email": email,
+            "password": password
+        }
+        headers = {
+            'X-STF-VALIDATION': self.recaptcha
+        }
+        response = self.session.post(url, json=payload, headers=headers)
+        response.raise_for_status()
+        return response.json()
+
+    def finalize_signup(self, signup_token):
+        url = f"{self.base_url}/api/auth/magic?token={signup_token}"
+        response = self.session.post(url)
+        response.raise_for_status()
+        return response.json()
+
+    def set_access_token(self, access_token):
+        self.access_token = access_token
+
+    def send_onboard_inquiry(self, data=TEST_ONBOARD_INQUIRY):
+        url = f"{self.base_url}/api/v1/user/onboarding_inquiry"
+        response = self.session.put(url, json=data, headers=self.auth_headers)
+        response.raise_for_status()
+
+    def create_account(self, slug):
+        if not self.access_token:
+            raise ValueError("Access token is not set. Please provide an access token.")
+        url = f"{self.base_url}/api/v1/accounts"
+        response = self.session.post(url, json={"slug": slug, "title": slug}, headers=self.auth_headers)
+        response.raise_for_status()
+        return response.json()["id"]
+
+
+@pytest.fixture(scope="function")
+def global_api():
+    global_config = CLUSTERS_CONFIG[TEST_ENV]["global"]
+    return GlobalAPI(global_config["base_url"], global_config["recaptcha"], global_config["root_pat_token"])
```

Comment:
Why not having a variable that you can import instead of a fixture?

```
global_config = CLUSTERS_CONFIG[TEST_ENV]["global"]
global_api = GlobalAPI(global_config["base_url"], global_config["recaptcha"], global_config["root_pat_token"])
```

Also, in case you prefer to keep it as a fixture, shouldn't the scope be `session`?

Reply:
it's just a matter of having everyhing setup related following the same patterns, i know the contents here are pretty static.